vllm/model_executor/layers/linear.py

```diff
@@ -573,11 +574,13 @@ def weight_loader_v2(self,
                          param: BasevLLMParameter,
                          loaded_weight: torch.Tensor,
                          loaded_shard_id: Optional[int] = None):
-        param_data = param.data
         if loaded_shard_id is None:
-            if param.output_dim is None:
-                assert param_data.shape == loaded_weight.shape
-                param_data.copy_(loaded_weight)
+            if isinstance(param, PerTensorScaleParameter):
+                param.load_merged_column_weight(loaded_weight=loaded_weight,
+                                                shard_id=0)
+                return
+            elif type(param) is BasevLLMParameter:
```

Comment:
switch to use isinstance

Reply:
we dont want `isinstance` in this case. all the parameters we introduced as instances of `BasevLLMParameter` so they'll all return True for `instance`. Specifically targeting parameters of the type `BasevLLMParameter`

Reply:
I see, okay